src/extension/coqPilot.ts

```diff
@@ -30,44 +27,41 @@ import { Uri } from "../utils/uri";
 
 import {
     buildTheoremsRankerFromConfig,
-    parseCoqLspServerPath,
     readAndValidateUserModelsParams,
 } from "./configReaders";
 import {
     deleteTextFromRange,
     highlightTextInEditor,
     insertCompletion,
 } from "./documentEditor";
-import { suggestAddingAuxFilesToGitignore } from "./editGitignoreCommand";
 import {
     EditorMessages,
     showMessageToUser,
     showMessageToUserWithSettingsHint,
 } from "./editorMessages";
+import { CompletionAbortError, throwOnAbort } from "./extensionAbortUtils";
 import { GlobalExtensionState } from "./globalExtensionState";
+import { HealthStatusIndicator } from "./healthStatusIndicator";
 import { subscribeToHandleLLMServicesEvents } from "./llmServicesEventsHandler";
 import {
     positionInRange,
     toVSCodePosition,
     toVSCodeRange,
 } from "./positionRangeUtils";
+import { SessionExtensionState } from "./sessionExtensionState";
 import { SettingsValidationError } from "./settingsValidationError";
-import { cleanAuxFiles, hideAuxFiles } from "./tmpFilesCleanup";
 
 export const pluginId = "coqpilot";
 export const pluginName = "CoqPilot";
 
 export class CoqPilot {
-    private readonly globalExtensionState: GlobalExtensionState;
-    private readonly vscodeExtensionContext: ExtensionContext;
-
-    constructor(vscodeExtensionContext: ExtensionContext) {
-        hideAuxFiles();
-        suggestAddingAuxFilesToGitignore();
-
-        this.vscodeExtensionContext = vscodeExtensionContext;
-        this.globalExtensionState = new GlobalExtensionState();
+    private readonly healthStatusIndicator: HealthStatusIndicator;
 
+    private constructor(
+        private readonly vscodeExtensionContext: ExtensionContext,
```

Comment:
The contexts/states should be renamed for sure, now we have enough of them to get confused. For example:
* `vscodeExtensionContext: ExtensionContext` -> `vscodeContext: VSCodeContext`
* `globalExtensionState: GlobalExtensionState` -> `pluginState: PluginState`
* `sessionExtensionState: SessionExtensionState` -> `sessionState: SessionState`

Reply:
Or maybe the global one should become `context`, if there is nothing that changes regularly